src/modinv32_impl.h

```diff
@@ -15,49 +15,153 @@ static void secp256k1_modinv32_normalize_30(secp256k1_modinv32_signed30 *r, int3
     const int32_t M30 = (int32_t)(UINT32_MAX >> 2);
     int32_t r0 = r->v[0], r1 = r->v[1], r2 = r->v[2], r3 = r->v[3], r4 = r->v[4],
             r5 = r->v[5], r6 = r->v[6], r7 = r->v[7], r8 = r->v[8];
-    int32_t c, cond_add;
+    int32_t cond_add;
 
     cond_add = r8 >> 31;
 
-    c  = r0 + (modinfo->modulus.v[0] & cond_add);
-    r0 = c & M30; c >>= 30;
-    c += r1 + (modinfo->modulus.v[1] & cond_add);
-    r1 = c & M30; c >>= 30;
-    c += r2 + (modinfo->modulus.v[2] & cond_add);
-    r2 = c & M30; c >>= 30;
-    c += r3 + (modinfo->modulus.v[3] & cond_add);
-    r3 = c & M30; c >>= 30;
-    c += r4 + (modinfo->modulus.v[4] & cond_add);
-    r4 = c & M30; c >>= 30;
-    c += r5 + (modinfo->modulus.v[5] & cond_add);
-    r5 = c & M30; c >>= 30;
-    c += r6 + (modinfo->modulus.v[6] & cond_add);
-    r6 = c & M30; c >>= 30;
-    c += r7 + (modinfo->modulus.v[7] & cond_add);
-    r7 = c & M30; c >>= 30;
-    c += r8 + (modinfo->modulus.v[8] & cond_add);
-    r8 = c;
-
-    cond_add = (c >> 31) ^ cond_negate;
-
-    c  = (r0 ^ cond_negate) - cond_negate + (modinfo->modulus.v[0] & cond_add);
-    r->v[0] = c & M30; c >>= 30;
-    c += (r1 ^ cond_negate) - cond_negate + (modinfo->modulus.v[1] & cond_add);
-    r->v[1] = c & M30; c >>= 30;
-    c += (r2 ^ cond_negate) - cond_negate + (modinfo->modulus.v[2] & cond_add);
-    r->v[2] = c & M30; c >>= 30;
-    c += (r3 ^ cond_negate) - cond_negate + (modinfo->modulus.v[3] & cond_add);
-    r->v[3] = c & M30; c >>= 30;
-    c += (r4 ^ cond_negate) - cond_negate + (modinfo->modulus.v[4] & cond_add);
-    r->v[4] = c & M30; c >>= 30;
-    c += (r5 ^ cond_negate) - cond_negate + (modinfo->modulus.v[5] & cond_add);
-    r->v[5] = c & M30; c >>= 30;
-    c += (r6 ^ cond_negate) - cond_negate + (modinfo->modulus.v[6] & cond_add);
-    r->v[6] = c & M30; c >>= 30;
-    c += (r7 ^ cond_negate) - cond_negate + (modinfo->modulus.v[7] & cond_add);
-    r->v[7] = c & M30; c >>= 30;
-    c += (r8 ^ cond_negate) - cond_negate + (modinfo->modulus.v[8] & cond_add);
-    r->v[8] = c;
+    r0 += modinfo->modulus.v[0] & cond_add;
+    r1 += modinfo->modulus.v[1] & cond_add;
+    r2 += modinfo->modulus.v[2] & cond_add;
+    r3 += modinfo->modulus.v[3] & cond_add;
+    r4 += modinfo->modulus.v[4] & cond_add;
+    r5 += modinfo->modulus.v[5] & cond_add;
+    r6 += modinfo->modulus.v[6] & cond_add;
+    r7 += modinfo->modulus.v[7] & cond_add;
+    r8 += modinfo->modulus.v[8] & cond_add;
+
+    r0 = (r0 ^ cond_negate) - cond_negate;
```

Comment:
Any particular reason to split these lines off from the above block?

Reply:
Not sure I understand the question; is it just formatting?. I guess just for clarity I wrote it as - conditionally add modulus, conditionally negate, carry, conditionally add modulus, carry. Please adjust as you see fit.

Reply:
I guess my question was did you pick this just because of style, or because it results in more performant code (it looked like something that could affect ILP e.g.)? I guess the former.

Reply:
Yeah, I didn't experiment with performance in that way, though removing the accumulator `c` hopefully made the ILP clearer to the compiler. In theory any further rearrangement is just a matter of style, but I haven't checked whether that's true in practice.